### ISSUE TITLE
Trouble loading plugins in Viewer demo

### ISSUE THREAD

**Issue by tomlankhorst** [mosra/magnum-examples#96]
The maintainers of this project are doing a formidable job. 😄

I am trying to get the `viewer` example going (2020.06). 

I installed magnum the `vcpkg` way, I use CMake on Ubuntu 20.04. 
Then, I installed the magnum plugins too `vcpkg install magnum-plugins\[\*\]`. 

Now, I see that these plugin libraries are present in the `lib` dir of `vcpkg`, although static `.a`. 

When I try to run the example, I run into the following issue: 

```
PluginManager::Manager::Manager(): none of the plugin search paths in {/home/me/vcpkg/installed/x64-linux/debug/bin/magnum-d/importers, ../lib/magnum-d/importers, magnum-d/importers} exists and pluginDirectory was not set, skipping plugin discovery
PluginManager::Manager::load(): plugin AnySceneImporter is not static and was not found in
```
I have tried passing the `vcpkg` prefix to the plugin manager. 
I feel like I am missing a step, how do I get dynamic plugins working the `vcpkg` way? 

Thanks in advance for your time.

**Comment by tomlankhorst** [comment]
Now, it seems like the `vcpkg` `magnum-plugins` are static by default. 
I tried the _static plugins method_ and stumbled upon an issue where CMake couldn't find `Assimp`. 
As I understand it, assimp is to be found with `find_package(assimp)`, lower-case. 
As a work-around, I copied the `FindAssimp.cmake` file from [magnum-plugins/modules](https://github.com/mosra/magnum-plugins/tree/master/modules), but shouldn't this be done automatically? 

Also, what's the way to use dynamic modules with `vcpkg`?

**Comment by mosra** [comment]
Hi!

Not exactly sure what vcpkg does on linux/macOS, on Windows at least you have an option to build a dynamic or a static version, and with the dynamic version things should magically "just work". I vaguely remember some update where vcpkg stopped offering dynamic builds on non-Windows platforms? In that case you'd be out of luck, since all examples assume dynamic plugins (as that's the default, and a much more flexible workflow since you're not tied to a particular plugin but "whatever can open PNGs" instead, for example).

For static plugins you'd need to explicitly link those (as you did), and the `Find` modules currently have to be copied, yes (I can't just install the 3rd party Find modules on the system to avoid conflicts, but I plan on fixing that somehow for the next release -- mosra/magnum#453).

In general, I'd suggest to avoid using vcpkg unless you're on Windows, as the other systems have far better package managers that don't try to force silly restrictions like vcpkg does, not to mention vcpkg upgrades have always been extremely painful for everybody. On Ubuntu [you're encouraged to build & install the .deb packages](https://doc.magnum.graphics/magnum/building.html#building-packages-deb) :)

**Comment by mosra** [comment]
Update -- I got another report related to this today, and after digging a bit I realized what's the culprit -- for dynamic plugins to work, these two lines https://github.com/microsoft/vcpkg/blob/b759049a36728d18260963799a56e6b19cb4a2ef/ports/magnum-plugins/portfile.cmake#L102-L103 should be wrapped in `if(WIN32)`. If you do that change locally in your vcpkg copy, dynamic plugins should get installed properly.

We have one vcpkg PR pending right now, when that one is merged I'll submit a fix for this. Will keep you updated.
